tests/std/tests/P2321R2_views_zip_transform/test.cpp

```diff
@@ -0,0 +1,864 @@
+    using range_type = test::range<Category, Element, IsSized, Diff, IsCommon,
+        (test::to_bool(IsCommon) ? test::CanCompare::yes
+                                 : test::CanCompare{derived_from<Category, forward_iterator_tag>}),
+        test::ProxyRef{derived_from<Category, contiguous_iterator_tag>}, test::CanView::yes,
+        test::Copyability::move_only>;
+};
+
+template <>
+class range_type_solver<false> {
+protected:
+    template <class Category, class Element, test::Sized IsSized, test::Common IsCommon, test::CanDifference Diff>
+    using range_type = test::range<Category, Element, IsSized, Diff, IsCommon,
+        (test::to_bool(IsCommon) ? test::CanCompare::yes
+                                 : test::CanCompare{derived_from<Category, forward_iterator_tag>})>;
+};
+
+template <bool IsMoveOnly, class Category, test::Sized IsSized, test::Common IsCommon, test::CanDifference Diff>
+class instantiator_impl : private range_type_solver<IsMoveOnly> {
+private:
+    template <class OtherCategory, class Element, test::Sized OtherIsSized, test::Common OtherIsCommon,
+        test::CanDifference OtherDiff>
+    using range_type = typename range_type_solver<IsMoveOnly>::template range_type<OtherCategory, Element, OtherIsSized,
+        OtherIsCommon, OtherDiff>;
+
+    using standard_range_type = range_type<Category, const int, IsSized, IsCommon, Diff>;
+
+    using differing_category_range_type =
+        range_type<conditional_t<same_as<Category, input_iterator_tag>, forward_iterator_tag, input_iterator_tag>,
+            const int, IsSized, IsCommon, Diff>;
+    using differing_size_member_range_type            = range_type<Category, const int,
+        (IsSized == test::Sized::yes ? test::Sized::no : test::Sized::yes), IsCommon, Diff>;
+    using differing_is_common_range_type              = range_type<Category, const int, IsSized,
+        (IsCommon == test::Common::yes ? test::Common::no : test::Common::yes), Diff>;
+    using differing_iterator_sentinel_diff_range_type = range_type<Category, const int, IsSized, IsCommon,
+        (Diff == test::CanDifference::yes ? test::CanDifference::no : test::CanDifference::yes)>;
+
+    static constexpr void test_single_range() {
+        auto single_range = standard_range_type{span<const int>{test_element_array_one}};
+        test_one(one_element_transform_closure, single_range_transform_results_array, single_range);
+    }
+
+    template <class DifferingRangeType>
+    static constexpr void test_three_ranges() {
+        auto first_range  = DifferingRangeType{span<const int>{test_element_array_one}};
+        auto second_range = standard_range_type{span<const int>{test_element_array_two}};
+        auto third_range  = standard_range_type{span<const int>{test_element_array_three}};
+
+        test_one(three_element_transform_closure, three_range_transform_results_array, first_range, second_range,
+            third_range);
+    }

```

Comment:
These `static` member functions are using the "almost always auto" pattern that we conventionally avoid. It is simpler to say `Type variable{initializer}` instead of `auto variable = Type{initializer}`, and reserves `auto` for the cases where type deduction is actually necessary.